mlir/include/mlir/IR/DialectImplementation.h

```diff
@@ -63,6 +78,9 @@ struct FieldParser<
     AttributeT, std::enable_if_t<std::is_base_of<Attribute, AttributeT>::value,
                                  AttributeT>> {
   static FailureOr<AttributeT> parse(AsmParser &parser) {
+    if constexpr (HasStaticDialectName<AttributeT>::value) {
+      parser.getContext()->getOrLoadDialect(AttributeT::dialectName);
+    }
```

Comment:
Should we complain when the field is missing? This would be the case for attributes defined outside tablegen.

Reply:
You mean a diagnostic? It seems that quite a few builtin attributes have this property (BoolAttr, e.g.,), so I'm not sure what one could do if they saw such a diagnostic... If we want to ensure all attributes have a dialectName string for this purpose, we could add it as a virtual method on `Attribute`. I don't have a strong opinion, I just tried to do the least invasive solution I could manage.